src/Core/ServiceProvider.php

```diff
@@ -43,8 +43,17 @@ function () {
                 if (!function_exists('WP_Filesystem')) {
                     require_once ABSPATH . '/wp-admin/includes/file.php';
                 }
+                $args = [];
+                if (get_option('ftp_credentials')) {
+                    $credentials = get_option('ftp_credentials');
+                    $args = [
+                        'hostname' => $credentials['hostname'],
+                        'username' => $credentials['username'],
+                        'password' => $credentials['password'],
+                    ];
+                }
```

Comment:
```suggestion
                $ftpCredentials = get_option('ftp_credentials');
                if ($ftpCredentials) {
                    $args = [
                        // TODO Define keys as Constants
                        'hostname' => array_key_exists('hostname', $ftpCredentials) ? $ftpCredentials['hostname'] : '',
                        'username' => array_key_exists('username', $ftpCredentials) ? $ftpCredentials['username'] : '',
                        'password' => array_key_exists('password', $ftpCredentials) ? $ftpCredentials['password'] : '',
                    ];
                }
```

I call `array_key_exists` because I'm not interested in value, just to ensure the key exists to prevent possible undefined index issues.